### PR TITLE
[FR] Update to 2.1-preview.2 with generation and recordings

### PR DESCRIPTION
**Pull request by maririos**
- Expose new features from service as is.
- I had to change some tests because service output changed.
- The service hasn't deployed receipts and BC in their 2.1-preview.2 endpoints so disable those tests (for now).
- Re-record all tests

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 1]
```diff
@@ -43,6 +43,7 @@ public void FormRecognizerClientCannotAuthenticateWithFakeApiKey()
         }
 
         [Test]
+        [Ignore("ADD not supported")]
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:46]: Should it be AAD?

By the way, are we not supporting AAD anymore for FR?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 2]
```diff
@@ -142,22 +143,31 @@ public async Task StartRecognizeContentPopulatesFormPagePdf(bool useStream)
...
+                // Row = 1 and columns 0-4 have a row span of 2.
+                var expectedRowSpan = (cell.RowIndex == 1 && cell.ColumnIndex != 5) ? 2 : 1;
+                Assert.AreEqual(expectedRowSpan, cell.RowSpan, $"Cell with text {cell.Text} should have a row span of {expectedRowSpan}.");
 
                 Assert.IsFalse(cell.IsFooter, $"Cell with text {cell.Text} should not have been classified as footer.");
                 Assert.IsFalse(cell.IsHeader, $"Cell with text {cell.Text} should not have been classified as header.");
 
                 Assert.GreaterOrEqual(cell.Confidence, 0, $"Cell with text {cell.Text} should have confidence greater or equal to zero.");
-                Assert.LessOrEqual(cell.RowIndex, 1, $"Cell with text {cell.Text} should have a row index less than or equal to one.");
+                Assert.LessOrEqual(cell.RowIndex, 2, $"Cell with text {cell.Text} should have a row index less than or equal to one.");
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:158]:
```suggestion
                Assert.LessOrEqual(cell.RowIndex, 2, $"Cell with text {cell.Text} should have a row index less than or equal to two.");
```

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 3]
```diff
@@ -142,22 +143,31 @@ public async Task StartRecognizeContentPopulatesFormPagePdf(bool useStream)
...
-                Assert.Greater(cell.FieldElements.Count, 0, $"Cell with text {cell.Text} should have at least one field element.");
+                // row = 2, column = 5 has empty text and no elements
+                if (cell.RowIndex == 2 && cell.ColumnIndex == 5)
+                {
+                    Assert.IsEmpty(cell.Text);
+                    Assert.AreEqual(0, cell.FieldElements.Count);
+                }
+                else
+                {
+                    Assert.AreEqual(expectedText[cell.RowIndex, cell.ColumnIndex], cell.Text);
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:168]: I'm a bit surprised this works. `expectedText` has dimensions [2, 6], but we get back cells with rowIndex = 2. Shouldn't we get an `ArgumentOutOfRangeException` here?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 4]
```diff
@@ -851,6 +874,7 @@ public async Task StartRecognizeReceiptsWithSupportedLocale(string locale)
         }
 
         [Test]
+        [Ignore("Not supported")]
         public void StartRecognizeReceiptsWithWrongLocale()
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:878]: Why do we have so many `Not supported`? Are we not supporting receipts and business cards anymore?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Models/OperationsLiveTests.cs [hunk 5]
```diff
@@ -106,6 +107,7 @@ public async Task TrainingOperationCanPollFromNewObject()
         }
 
         [Test]
+        [Ignore("CHECK REGIONS")]
         public async Task CopyModelOperationCanPollFromNewObject()
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Models/OperationsLiveTests.cs:111]: Check regions?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/api/Azure.AI.FormRecognizer.netstandard2.0.cs [hunk 6]
```diff
@@ -333,6 +334,24 @@ public enum SelectionMarkState
         Selected = 0,
         Unselected = 1,
     }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct TextStyle : System.IEquatable<Azure.AI.FormRecognizer.Models.TextStyle>
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/api/Azure.AI.FormRecognizer.netstandard2.0.cs:338]: It seems we added this struct to the public API, but where is it being used?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/autorest.md [hunk 7]
```diff
@@ -7,12 +7,10 @@ Run `dotnet build /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    -  https://github.com/Azure/azure-rest-api-specs/blob/8fee11f5dd30e228a850ee25a2f3368dc1671b53/specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.1-preview.1/FormRecognizer.json
+    - .\swagger\fr2.1p2.json
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/autorest.md:10]: Are we keeping a temporary local swagger while the service team does not update it on their side?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs (hunk 3 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:168]: hmm I see the confusion. So the service only returns one cell for `rowIndex=2` which is `(2,5)`. do you think I should be more explicit?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs (hunk 4 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:878]: The service hasn't deployed receipts and BC in their 2.1-preview.2 endpoints so we decided to work on a feature branch and disable those tests.
I will add this to the description of the PR.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs (hunk 1 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:46]: lol. yes. thanks!
2.1-preview.2 doesn't support AAD yet so decided to ignore the test until the service can fix it

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Models/OperationsLiveTests.cs (hunk 5 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Models/OperationsLiveTests.cs:111]: Changos!!! this was a note for me to check this test again. Thank you!!!

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs (hunk 3 above)
Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:168]: I see. It's just testing, so I'm not that concerned.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/autorest.md (hunk 7 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/autorest.md:10]: yeah... they have a PR open but not much progress so we internally are using this swagger to get unblock and develop new features.
This work is only in feature branch. I will only move to master when the service team merges their PR

File: sdk/formrecognizer/Azure.AI.FormRecognizer/api/Azure.AI.FormRecognizer.netstandard2.0.cs (hunk 6 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/api/Azure.AI.FormRecognizer.netstandard2.0.cs:338]: Oh I missed this one. I will hide

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs (hunk 4 above)
Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:878]: I see. I hadn't noticed this was a feature branch.

**Review by kinelski — APPROVED** [review]
Hadn't noticed this was a feature branch. LGTM!

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Models/OperationsLiveTests.cs (hunk 5 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Models/OperationsLiveTests.cs]: Enabled again

**Pull request merged**